src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1822,8 +1821,13 @@ static void perform_transport_op_locked(grpc_exec_ctx* exec_ctx,
         op->on_connectivity_state_change);
   }
 
-  if (close_transport != GRPC_ERROR_NONE) {
-    close_transport_locked(exec_ctx, t, close_transport);
+  if (op->disconnect_with_error != GRPC_ERROR_NONE) {
+    close_transport_locked(exec_ctx, t, op->disconnect_with_error);
+    if (t->notify_on_receive_settings != nullptr) {
```

Comment:
why isn't this in close_transport_locked?

Reply:
Yeah, that does look like a better place for it.  Done.